cloud/cloudstack/cs_ip_address.py

```diff
@@ -0,0 +1,459 @@
+
+    def poll_job(self, job=None, key=None):
+        if 'jobid' in job:
+            while True:
+                res = self.cs.queryAsyncJobResult(jobid=job['jobid'])
+                if res['jobstatus'] != 0 and 'jobresult' in res:
+                    if 'errortext' in res['jobresult']:
+                        self.module.fail_json(msg="Failed: '%s'" % res['jobresult']['errortext'])
+                    if key and key in res['jobresult']:
+                        job = res['jobresult'][key]
+                    break
+                time.sleep(2)
+        return job
+
+
+class AnsibleCloudStackIPAddress(AnsibleCloudStack):
+
+    def __init__(self, module):
+        AnsibleCloudStack.__init__(self, module)
+        self.vm_default_nic = None
+
+    def associate_ip_address(self):
+        self.result['changed'] = True
+        args = {}
+        args['account'] = self.get_account(key='id')
+        args['networkid'] = self.get_network(key='id')
+        args['zoneid'] = self.get_zone(key='id')
+        ip_address = {}
+        if not self.module.check_mode:
+            res = self.cs.associateIpAddress(**args)
+            if 'errortext' in res:
+                self.module.fail_json(msg="Failed: '%s'" % res['errortext'])
+
+            poll_async = self.module.params.get('poll_async')
+            if poll_async:
+                res = self._poll_job(res, 'ipaddress')
+            ip_address = res
+        return ip_address
+
+    def disassociate_ip_address(self):
+        ip_address = self.get_ip_address()
+        if ip_address is None:
+            return ip_address
+        if ip_address['isstaticnat']:
+            self.module.fail_json(msg="IP address is allocated via static nat")
+
+        self.result['changed'] = True
+        if not self.module.check_mode:
+            res = self.cs.disassociateIpAddress(id=ip_address['id'])
+            if 'errortext' in res:
+                self.module.fail_json(msg="Failed: '%s'" % res['errortext'])
+            poll_async = self.module.params.get('poll_async')
+            if poll_async:
+                res = self._poll_job(res, 'ipaddress')
+        return ip_address
+
+    def get_result(self, ip_address):
+        if ip_address:
+            if 'zonename' in ip_address:
+                self.result['zone'] = ip_address['zonename']
+            if 'domain' in ip_address:
+                self.result['domain'] = ip_address['domain']
+            if 'account' in ip_address:
+                self.result['account'] = ip_address['account']
+            if 'project' in ip_address:
+                self.result['project'] = ip_address['project']
+            if 'ipaddress' in ip_address:
+                self.result['ip_address'] = ip_address['ipaddress']
+            if 'id' in ip_address:
+                self.result['id'] = ip_address['id']
+        return self.result
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            ip_address = dict(required=False),
```

Comment:
It's not required for state=present, only state=absent. Can I configure that on the module?